app/controllers/posts_controller.rb

```diff
@@ -18,49 +20,41 @@ def edit
   # POST /posts
   def create
     @post = Post.new(post_params)
+    @post.user = current_user
 
-    respond_to do |format|
-      if @post.save
-        format.html { redirect_to @post, notice: 'Post was successfully created.' }
-        # format.json { render :show, status: :created, location: @post }
-      else
-        format.html { render :new }
-        # format.json { render json: @post.errors, status: :unprocessable_entity }
-      end
+    if @post.save
+      redirect_to authenticated_root_path
+    else
+      flash.now[:error] = @post.errors.messages
+      render :new
     end
   end
 
   # PATCH/PUT /posts/1
   def update
     post
-    respond_to do |format|
-      if @post.update(post_params)
-        format.html { redirect_to @post, notice: 'Post was successfully updated.' }
-        # format.json { render :show, status: :ok, location: @post }
-      else
-        format.html { render :edit }
-        # format.json { render json: @post.errors, status: :unprocessable_entity }
-      end
+    if @post.update(post_params)
+      redirect_to @post, notice: 'Post was successfully updated.'
+    else
+      render :edit
     end
   end
 
   # DELETE /posts/1
   def destroy
     post
     @post.destroy
```

Comment:
no need to repeat it, magic of `||=` allows you to replace:

```
post
@post.destroy
```

with

```
post.destroy
```

as method `post` returns @post ;)